src/abstractdataframe/iteration.jl

```diff
@@ -454,7 +454,17 @@ function mapcols!(f::Union{Function, Type}, df::DataFrame)
     if len_min != len_max
         throw(DimensionMismatch("lengths of returned vectors must be identical"))
     end
-    _columns(df) .= vs
```

Comment:
What was the problem with this syntax? Compilation overhead?

Reply:
yes - I try to remove broadcasting in the internal code to reduce latency a bit (also I thought that the syntax might seem too magic for casual reader).

Reply:
The difference is:

current release:
```
julia> using DataFrames

julia> df = DataFrame(a=1)
1×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1

julia> @time mapcols!(length, df)
  0.003640 seconds (2.75 k allocations: 146.932 KiB)
1×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1
```
this PR:
```
julia> using DataFrames

julia> df = DataFrame(a=1)
1×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1

julia> @time mapcols!(length, df)
  0.000015 seconds (4 allocations: 256 bytes)
1×1 DataFrame
 Row │ a     
     │ Int64 
─────┼───────
   1 │     1
```
so it is not a big deal (but we have to iterate `vs` anyway to check `firstindex`).

Reply:
Wow. I wouldn't have expected such a large difference in the number of allocations.

Reply:
broadcasting is heavy (when doing the PRs that Tim Holy prompted I will review the whole codebase against using broadcasting). Of course here the difference is one-time and not that big.